### PR TITLE
[EOS-18467] Provisioner CLI to refresh machine_id

### PR DESCRIPTION
**Pull request by mazinamdar**
Signed-off-by: mazinamdar <mazhar.inamdar@seagate.com>

Forcefully refresh machine_id
`provisioner refresh_machine_id --forcce`

Normal refresh
`provisioner refresh_machine_id`

```
API to refresh machine id on all nodes

optional arguments:
  -h, --help            show this help message and exit
  --force               Forcefully update machine_id (default: False)

```

### CODE REVIEW

**Review by andkononykhin2 — APPROVED** [review]

File: api/python/provisioner/commands/refresh_machine_id.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            state = "components.provisioner.config.machine_id.force_refresh"
+        else:
+            logger.info("Refreshing machine id on all nodes")
+            state = "components.provisioner.config.machine_id.refresh_machine_id"  # noqa: E501
+
+        StatesApplier.apply([state])
+
+        logger.info("refresh grains on all nodes")
+        StatesApplier.apply(
+            ["components.provisioner.config.machine_id.refresh_grains"]
```
Comment (ypise) [inline api/python/provisioner/commands/refresh_machine_id.py:68]: Design change request:
```
    machine_id_refresh_states = []
    if kwargs['force']:
        logger.debug("machine_id on all nodes shall be forcefully reset.")
        machine_id_refresh_states.append("components.provisioner.config.machine_id.force_reset")
    else:
        logger.debug("machine_id on all nodes shall be reset.")
        machine_id_refresh_states.append("components.provisioner.config.machine_id.reset") 
   # noqa: E501

    logger.debug("Refresh grains on all nodes post machine_id reset.")
    machine_id_refresh_states.append("components.provisioner.config.machine_id.refresh_grains")

    logger.info("Resetting machine_id on all nodes.")
    StatesApplier.apply(
            ["components.provisioner.config.machine_id.refresh_grains"]
    )

    logger.info("SUCCESS: machine_id is reset.")
```

File: api/python/provisioner/api_spec.yaml [hunk 2]
```diff
@@ -115,6 +115,8 @@ salt:
   type: SaltClient
 generate_roster:
   type: GenerateRoster
+refresh_machine_id:
+  type: RefreshMachineId
```
Comment (ypise) [inline api/python/provisioner/api_spec.yaml:119]: Rename this to reset as we are actually re-setting the id.

File: srv/components/provisioner/config/machine_id/force_refresh.sls [hunk 3]
```diff
@@ -0,0 +1,39 @@
+#
```
Comment (ypise) [inline srv/components/provisioner/config/machine_id/force_refresh.sls:1]: Rename this file to `force-reset.sls`

**Review by ypise — CHANGES_REQUESTED** [review]

File: api/python/provisioner/commands/refresh_machine_id.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    @staticmethod
+    def run(**kwargs):
+
+        if kwargs['force']:
+            logger.info("Forcefully refresh machine id on all nodes")
+            state = "components.provisioner.config.machine_id.force_refresh"
+        else:
+            logger.info("Refreshing machine id on all nodes")
+            state = "components.provisioner.config.machine_id.refresh_machine_id"  # noqa: E501
```
[the remark this answers is not in the document]
Comment (mazinamdar) [reply api/python/provisioner/commands/refresh_machine_id.py]: comment is addressed

File: api/python/provisioner/commands/refresh_machine_id.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+        StatesApplier.apply([state])
+
+        logger.info("refresh grains on all nodes")
```
[the remark this answers is not in the document]
Comment (mazinamdar) [reply api/python/provisioner/commands/refresh_machine_id.py]: comment is addressed

File: api/python/provisioner/commands/refresh_machine_id.py (hunk 1 above)
Comment (mazinamdar) [reply api/python/provisioner/commands/refresh_machine_id.py]: comment is addressed

File: api/python/provisioner/api_spec.yaml (hunk 2 above)
Comment (mazinamdar) [reply api/python/provisioner/api_spec.yaml]: comment is addressed

File: srv/components/provisioner/config/machine_id/force_refresh.sls (hunk 3 above)
Comment (mazinamdar) [reply srv/components/provisioner/config/machine_id/force_refresh.sls]: comment is addressed

**Review by ypise — APPROVED** [review]

**Pull request merged**